compiler/compiler_test.go

```diff
@@ -49,7 +49,37 @@ func TestCompileAndEval(t *testing.T) {
 			}),
 			input: values.NewObjectWithValues(map[string]values.Value{
 				"r": values.NewObjectWithValues(map[string]values.Value{
-					"n": values.NewInt(10),
+					"n": values.NewString("2"),
+				}),
+			}),
+			want: values.NewString("n = 2"),
+		},
+		{
+			name: "interpolated string expression with duration type",
+			fn:   `(r) => "n = ${r.n}"`,
+			inType: semantic.NewObjectType([]semantic.PropertyType{
+				{Key: []byte("r"), Value: semantic.NewObjectType([]semantic.PropertyType{
+					{Key: []byte("n"), Value: semantic.BasicDuration},
+				})},
+			}),
+			input: values.NewObjectWithValues(map[string]values.Value{
+				"r": values.NewObjectWithValues(map[string]values.Value{
+					"n": values.NewString("1m"),
```

Comment:
Why is this is a `values.NewString`? I expected it to be a  `values.NewDuration`